package/network/config/ltq-vdsl-app/files/10_atm.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+[ "$DSL_NOTIFICATION_TYPE" = "DSL_STATUS" ] && \
+[ "$DSL_TC_LAYER_STATUS" = "ATM" ] && \
+! grep -q "ltq_atm_vr9" /proc/modules || exit 0
+
+logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer ATM"
+
+if grep -q "ltq_ptm_vr9" /proc/modules ; then
+	# TODO: make ltq_atm_vr9/ltq_ptm_vr9 "hot-swappable"
+	# ifconfig ptm0 down &>/dev/null
+	# rmmod "ltq_ptm_vr9"
+	logger -p daemon.notice -t "dsl-notify" "Loading ATM driver while EFM/PTM driver is loaded is not possible. Reboot is needed."
+	exit
+fi
+
+modprobe ltq_atm_vr9
+esi $(printf '%012X' $((1+0x$(tr -d : </sys/class/net/eth0/address))))
+
+grep -q "br2684" /proc/modules || modprobe br2684
```

Comment:
Hmmm, when i drop that, the atm part won't work anymore as i stated above:

> The current implementation (starting br2684 via init script) only works, because dsl_control (where the atm driver is loaded) is started before the br2684 init script.
>
>When the atm driver is loaded by the notify script, the br2684 init script would be called a long time ago and the br2684 won't be setup correctly.